### ISSUE TITLE
TS with eslint: Duplicate function implementation in lib.dom.d.ts

### ISSUE THREAD

**Issue by Pluto1010** [nuxt/typescript#307]
Hello!

I'm new to nuxtjs and I've tried to set it up for TypeScript like described on https://typescript.nuxtjs.org/guide/setup.html

I think I haven't understood something related to the config of eslint. Can someone tell me what this errors mean and how I can avoid them? As I understand there are linter rules thinking that something in the typescript library is wrong. I expect that this is a false alarm.

Whats happening here?

Thank you very much in advance (at least for reading) :)

```
$ npm run dev
...

ERROR  ERROR in /frontend/node_modules/typescript/lib/lib.dom.d.ts(19625,18):               nuxt:typescript 21:11:59
19625:18 Duplicate function implementation.
    19623 | declare function captureEvents(): void;
    19624 | declare function close(): void;
  > 19625 | declare function confirm(message?: string): boolean;
          |                  ^
    19626 | declare function departFocus(navigationReason: NavigationReason, origin: FocusNavigationOrigin): void;
    19627 | declare function focus(): void;
    19628 | declare function getComputedStyle(elt: Element, pseudoElt?: string | null): CSSStyleDeclaration;


ERROR  ERROR in /frontend/node_modules/typescript/lib/lib.dom.d.ts(19625,18):               nuxt:typescript 21:11:59
19625:18 This overload signature is not compatible with its implementation signature.
    19623 | declare function captureEvents(): void;
    19624 | declare function close(): void;
  > 19625 | declare function confirm(message?: string): boolean;
          |                  ^
    19626 | declare function departFocus(navigationReason: NavigationReason, origin: FocusNavigationOrigin): void;
    19627 | declare function focus(): void;
    19628 | declare function getComputedStyle(elt: Element, pseudoElt?: string | null): CSSStyleDeclaration;
```

packages.json: 
```javascript
{
  "name": "myapp",
  "version": "1.0.0",
  "description": "Frontend for myapp",
  "author": "Pluto1010",
  "private": true,
  "scripts": {
    "lint": "eslint --ext .ts,.js,.vue --ignore-path ./.gitignore .",
    "test": "jest",
    "dev": "nuxt-ts",
    "build": "nuxt-ts build",
    "generate": "nuxt-ts generate",
    "start": "nuxt-ts start"
  },
  "dependencies": {
    "@nuxt/typescript-runtime": "^0.3.10",
    "@nuxtjs/axios": "^5.3.6",
    "@nuxtjs/dotenv": "^1.4.0",
    "@nuxtjs/pwa": "^3.0.0-0",
    "aws-amplify": "^2.2.6",
    "aws-amplify-vue": "^1.1.4",
    "graphql": "^14.6.0",
    "nuxt": "^2.0.0"
  },
  "devDependencies": {
    "@nuxt/typescript-build": "^0.5.8",
    "@nuxtjs/eslint-config": "^2.0.0",
    "@nuxtjs/eslint-config-typescript": "^1.0.2",
    "@nuxtjs/eslint-module": "^1.0.0",
    "@nuxtjs/vuetify": "^1.0.0",
    "@types/graphql": "^14.5.0",
    "@types/zen-observable": "^0.8.0",
    "@vue/test-utils": "^1.0.0-beta.27",
    "babel-eslint": "^10.0.1",
    "babel-jest": "^24.1.0",
    "eslint": "^6.1.0",
    "eslint-config-prettier": "^6.10.0",
    "eslint-plugin-nuxt": ">=0.4.2",
    "eslint-plugin-prettier": "^3.1.2",
    "jest": "^24.1.0",
    "prettier": "^1.19.1",
    "vue-jest": "^4.0.0-0"
  }
}
```

**Comment by NickBolles** [comment]
That doesn't look like an eslint error, it looks like a typescript error. Can you include your tsconfig?

**Comment by Pluto1010** [comment]
Sure.

```javascript
// tsconfig.json
{
  "compilerOptions": {
    "target": "es2018",
    "module": "esnext",
    "moduleResolution": "node",
    "lib": [
      "esnext",
      "esnext.asynciterable",
      "dom"
    ],
    "esModuleInterop": true,
    "allowJs": true,
    "sourceMap": true,
    "strict": true,
    "noEmit": true,
    "baseUrl": ".",
    "paths": {
      "~/*": [
        "./*"
      ],
      "@/*": [
        "./*"
      ]
    },
    "types": [
      "@types/node",
      "@nuxt/types"
    ]
  },
  "exclude": [
    "node_modules"
  ]
}
```

**Comment by kevinmarrec** [comment]
@Pluto1010 

As @NickBolles said, it's a TypeScript error.
Somehow it kinda looks like you have 2 TypeScript versions installed in your repository somehow, it may be a broken `node_modules` setup or cached dependencies.

Try removing `node_modules` and your lock file `yarn.lock` or `package-lock.json` then reinstall dependencies.

---
Aside note :

```json
{
  "@nuxtjs/eslint-config": "^2.0.0",
  "@nuxtjs/eslint-config-typescript": "^1.0.2",
  "@nuxtjs/eslint-module": "^1.0.0"
}
```

`@nuxtjs/eslint-config-typescript` already depends on `@nuxtjs/eslint-config` so you can safely remove `@nuxtjs/eslint-config` from your `package.json`

**Comment by Pluto1010** [comment]
@kevinmarrec I've just tried that - still the same :(

- I have removed `eslint-config` from `packages.json`. 
- Deleted `node_modules` and `package-lock.json`. 
- Reinstalled by calling `npm install`.

**Comment by kevinmarrec** [comment]
@Pluto1010 As I can't debug further without proper code, simply share a repository with the issue I'll take a look :)

**Comment by kevinmarrec** [comment]
@Pluto1010 Any update ?
